### ISSUE TITLE
Error parsing import statement in HCL-JSON

### ISSUE THREAD

**Issue by aherbst-broad** [opentofu/opentofu#1652]
### OpenTofu Version

```shell
» tofu --version                                       
OpenTofu v1.7.1
on darwin_arm64
+ provider registry.terraform.io/hashicorp/azurerm v3.95.0
```

### OpenTofu Configuration Files

```
{
  "//": {
    "metadata": {
      "backend": "pg",
      "imports": {
        "azurerm_storage_account": [
          "storageAccount"
        ]
      },
      "stackName": "storageAccount1234",
      "version": "0.20.7"
    },
    "outputs": {
    }
  },
  "import": [
    {
      "id": "/subscriptions/cfb45937-0e77-4aaa-a3ec-d2d83e4df678/resourceGroups/test1234/providers/Microsoft.Storage/storageAccounts/storageAccount1234",
      "to": "azurerm_storage_account.storageAccount"
    }
  ],
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_storage_account": {
      "storageAccount": {
        "//": {
          "metadata": {
            "path": "storageAccount1234/storageAccount",
            "uniqueId": "storageAccount"
          }
        },
        "account_replication_type": "LRS",
        "account_tier": "Standard",
        "lifecycle": {
          "ignore_changes": [
            "location"
          ]
        },
        "location": "westus",
        "name": "lzstorageaccount1234",
        "resource_group_name": "storageAccount1234"
      }
    }
  },
  "terraform": {
    "backend": {
      "pg": {
        "conn_str": "postgres://postgres:mysecretpassword@localhost/terraform_backend?sslmode=disable"
      }
    },
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.95.0"
      }
    }
  }
}
```

### Debug Output

https://gist.github.com/aherbst-broad/4e7730390be12825dfcbac95c70f52c2

### Expected Behavior

The plan and subsequent apply should succeed instead of erroring out. Running against Terraform 1.8.3 succeeds.

### Actual Behavior

`tofu plan` errors out like this:

```
╷
│ Error: Invalid import address expression
│
│   on cdk.tf.json line 19, in import[0]:
│   19:       "to": "azurerm_storage_account.storageAccount"
│
│ Import address must be a reference to a resource's address, and only allows for indexing with dynamic keys. For example: module.my_module[expression1].aws_s3_bucket.my_buckets[expression2] for resources inside of modules, or simply
│ aws_s3_bucket.my_bucket for a resource in the root module
```

### Steps to Reproduce

1. `tofu init`
2. `tofu plan`

### References

May be related to this [commit](https://github.com/opentofu/opentofu/commit/63c88507a878513cf7f860eceefd024a6da78b57)

**Comment by ronnyorot** [comment]
Hi @aherbst-broad, and thank you for reporting this bug.
I managed to reproduce it using JSON files, it looks like we don't parse the `to` value correctly. The same configuration in TF files is working correctly.
We'll start working on a fix and update you when we are done. 
Thank you again for reporting this!

**Comment by Evi1Pumpkin** [comment]
Hi @aherbst-broad, and thank you for reporting this bug.
I managed to reproduce it using JSON files, it looks like we don't parse the to value correctly. The same configuration in TF files is working correctly.
We'll start working on a fix and update you when we are done.
Thank you again for reporting this!

**Comment by Evi1Pumpkin** [comment]
Hi @aherbst-broad,

We have fixed this bug in [version v1.7.2](https://github.com/opentofu/opentofu/releases/tag/v1.7.2), which is now available.

Thank you for your patience and for helping to improve OpenTofu. Your assistance is highly appreciated.
